From 6e95493628e6ec6bef4d9ec9122b67ca0351b2a0 Mon Sep 17 00:00:00 2001
From: Albert Astals Cid <aacid@kde.org>
Date: Mon, 21 Jan 2013 23:15:12 +0100
Subject: [PATCH 1/4] Search the gfortran library before using it
 unconditionally

Hope it fixes cantor build in build.kde.org
---
 cmake/FindR.cmake | 18 +++++++++++-------
 1 file changed, 11 insertions(+), 7 deletions(-)

diff --git a/cmake/FindR.cmake b/cmake/FindR.cmake
index 0022f90..63d0387 100644
--- a/cmake/FindR.cmake
+++ b/cmake/FindR.cmake
@@ -1,4 +1,4 @@
-# - Try to find R 
+# - Try to find R
 # Once done this will define
 #
 #  R_FOUND - system has R
@@ -40,7 +40,7 @@ IF(R_EXECUTABLE)
     MESSAGE(STATUS "R_Home not findable via R. Guessing")
   ENDIF(NOT R_INCLUDE_DIR)
 
-  FIND_PATH(R_INCLUDE_DIR R.h) 
+  FIND_PATH(R_INCLUDE_DIR R.h)
 
   # check for existence of libR.so
 
@@ -69,10 +69,14 @@ IF(R_EXECUTABLE)
     #MESSAGE(STATUS "Yes, ${R_LAPACK_LIBRARY} exists")
     SET(R_LIBRARIES ${R_LIBRARIES} ${R_LAPACK_LIBRARY})
     IF(NOT WIN32)
-      # needed when linking to Rlapack on linux for some unknown reason.
-      # apparently not needed on windows (let's see, when it comes back to bite us, though)
-      # and compiling on windows is hard enough even without requiring libgfortran, too.
-      SET(R_LIBRARIES ${R_LIBRARIES} gfortran)
+      FIND_LIBRARY(GFORTRAN_LIBRARY
+        gfortran)
+      if (GFORTRAN_LIBRARY)
+        # needed when linking to Rlapack on linux for some unknown reason.
+        # apparently not needed on windows (let's see, when it comes back to bite us, though)
+        # and compiling on windows is hard enough even without requiring libgfortran, too.
+        SET(R_LIBRARIES ${R_LIBRARIES} gfortran)
+      endif (GFORTRAN_LIBRARY)
     ENDIF(NOT WIN32)
   ENDIF(NOT R_LAPACK_LIBRARY)
 
@@ -94,7 +98,7 @@ IF(R_EXECUTABLE)
 ENDIF( R_EXECUTABLE )
 
 INCLUDE(FindPackageHandleStandardArgs)
-FIND_PACKAGE_HANDLE_STANDARD_ARGS(R  DEFAULT_MSG 
+FIND_PACKAGE_HANDLE_STANDARD_ARGS(R  DEFAULT_MSG
                                   R_EXECUTABLE R_INCLUDE_DIR R_R_LIBRARY)
 
 MARK_AS_ADVANCED(R_INCLUDE_DIR R_R_LIBRARY R_LAPACK_LIBRARY R_BLAS_LIBRARY)
-- 
1.8.1

